tests/unit/help_test.sh

```diff
@@ -32,4 +32,74 @@ function test_kworkflow_man()
   assertEquals "($LINENO) We expected an error message." "$expect" "$output"
 }
 
+function test_kworkflow_version()
+{
+  local KW_LIB_DIR
+  local output
+  local expected_output
+
+  # the mocked version for KW
+  expected_output=$(printf 'beta\nBranch: bazz\nCommit: ffddee\n')
+
+  # this value mocks a fake location for the version file, which is the file used
+  # by KW to get information about the current version.
+  KW_LIB_DIR="${SHUNIT_TMPDIR}"
+  printf '%s' "${expected_output}" > "${KW_LIB_DIR}/VERSION"
+
+  # check output
+  output=$(kworkflow_version)
+  assertEquals "(${LINENO}) Got wrong kw version." "${expected_output}" "${output}"
+}
+
+function test_kworkflow_version_in_repomode()
+{
+  local branch_name
+  local head_hash
+  local version_name
+  local output
+  local path_to_git_repository
+  local KW_REPO_MODE
+  local KW_LIB_DIR
+
+  # Initialize variables.
+  version_name='omega'
+  branch_name='foo'
+  path_to_git_repository="${SHUNIT_TMPDIR}/git_repo"
+  KW_LIB_DIR="${path_to_git_repository}/src"
+
+  # shellcheck disable=SC2034
+  KW_REPO_MODE='y' # used to trigger repo-mode behavior
+
+  # create the git repository
+  mkdir --parents "${path_to_git_repository}" "${KW_LIB_DIR}"
+
+  # cd into the repo, so we can run git commands there.
+  # This is necessary because our helper `mk_fake_git` assumes the current
+  # working directory is the directory in which we want to run git commands.
+  cd "${path_to_git_repository}" || fail 'Failed to switch to git dir.'
```

Comment:
Before this line, I think it is cleaner to store the original path as `original_path="$PWD"` (don't forget to declare it as local) and use this value to go back to the starting dir at the end of the test.